packages/core/src/login/webview/vue/amazonq/backend_amazonq.ts

```diff
@@ -56,7 +56,13 @@ export class AmazonQLoginWebview extends CommonAuthWebview {
                 isReAuth: false,
             })
 
-            await awsIdSignIn()
+            const conn = await awsIdSignIn()
+            const registration = await conn.getRegistration()
+            this.storeMetricMetadata({
+                ssoRegistrationExpiresAt: registration?.expiresAt.toISOString(),
```

Comment:
There is some duplication of storing registration in the metric metadata. Would it make sense for a generalized function that takes in the registration as an arg and that adds it to the metadata? This file and backend_toolkit could reuse this